kernel_hardening_checker/engine.py

```diff
@@ -98,20 +98,20 @@ def check(self):
         else:
             self.result = f'FAIL: "{self.state}"'
 
-    def table_print(self, _mode, with_results):
+    def table_print(self, _mode, with_results: bool):
         print(f'{self.name:<40}|{self.type:^7}|{self.expected:^12}|{self.decision:^10}|{self.reason:^18}', end='')
         if with_results:
             print(f'| {colorize_result(self.result)}', end='')
 
-    def json_dump(self, with_results):
+    def json_dump(self, with_results: bool) ->dict:
         dump = {
             "option_name": self.name,
             "type": self.type,
             "desired_val": self.expected,
             "decision": self.decision,
             "reason": self.reason,
         }
-        if with_results:
+        if with_results and self.result:
```

Comment:
A question about this line.
All other changes add types in the function prototypes, but here you change the code.
Could you describe the reason?

Reply:
> kernel_hardening_checker/engine.py:119: error: "None" has no attribute "startswith"  [attr-defined]